markup.go

```diff
@@ -154,26 +151,43 @@ func (m ClassMap) Apply(h *HTML) {
 	Property("className", strings.Join(classes, " ")).Apply(h)
 }
 
-// List represents a list of Markup, Component, or HTML which is individually
+// markupList represents a list of Applyer which is individually
 // applied to an HTML element or text node.
-type List []MarkupOrComponentOrHTML
+type markupList []Applyer
 
-// Apply implements the Markup interface.
-func (l List) Apply(h *HTML) {
-	for _, m := range l {
-		apply(m, h)
+// Apply implements the Applyer interface.
+func (m markupList) Apply(h *HTML) {
+	for _, a := range m {
+		if a == nil {
+			continue
+		}
+		a.Apply(h)
 	}
 }
 
+// Markup wraps a list of Applyer which is individually
+// applied to an HTML element or text node.
+func Markup(m ...Applyer) markupList { // nolint: golint
+	return markupList(m)
+}
+
 // If returns nil if cond is false, otherwise it returns the given markup.
-func If(cond bool, markup ...MarkupOrComponentOrHTML) MarkupOrComponentOrHTML {
+func If(cond bool, markup ...ComponentOrHTML) MarkupOrChild {
 	if cond {
 		return List(markup)
 	}
 	return nil
 }
 
-// UnsafeHTML is Markup which unsafely sets the inner HTML of an HTML element.
+// MarkupIf returns nil if cond is false, otherwise it returns the given markup.
+func MarkupIf(cond bool, markup ...Applyer) Applyer {
```

Comment:
Still an open question to me if there is a better API than this. But I won't block this PR on that, it's something we can improve in the future if we find a better alternative.